pkg/statistics/handle/autoanalyze/priorityqueue/interval.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package priorityqueue
+
+import (
+	"time"
+
+	"github.com/pingcap/tidb/pkg/sessionctx"
+	"github.com/pingcap/tidb/pkg/statistics/handle/util"
+	"github.com/pingcap/tidb/pkg/types"
+)
+
+const avgDurationQueryForTable = `
+	SELECT AVG(TIMESTAMPDIFF(SECOND, start_time, end_time)) AS avg_duration
+	FROM mysql.analyze_jobs
+	WHERE table_schema = %? AND table_name = %? AND state = 'finished' AND fail_reason IS NULL AND partition_name = ''
+	ORDER BY id DESC
+	LIMIT 5;
```

Comment:
This SQL is a bit questionable with both an `avg` and `ORDER BY + LIMIT`.

Reply:
I want to calculate the average analysis time for the most recent 5 records. Could you please provide more information about the issue that is causing you worry?

Reply:
The semantic of a SELECT statement is generally `JOIN -> WHERE -> AGG -> HAVING -> ORDER BY -> LIMIT`. You would need to write it like `avgDurationQueryForPartition` below to get what you want.
You can see this example: https://www.db-fiddle.com/f/3moV1YrnrZchZTaK41fk2h/1